python/tank/templatekey.py

```diff
@@ -292,31 +304,164 @@ def __init__(self,
                                          abstract=abstract,
                                          length=length)
 
-        if not(format_spec is None or isinstance(format_spec, basestring)):
-            msg = "Format_spec for TemplateKey %s is not of type string: %s"
+    def _init_format_spec(self, name, format_spec):
+        """
+        Asserts that the format_spec parameter is a valid value.
+
+        :param name: Name of this template key.
+        :param format_spec: Parameter to be validated.
+
+        :raises TankError: Raised when the parameter is not a string that maching a %d format
+                           option.
+        """
+        # No format spec means no formatting options.
+        if format_spec is None:
+            self._padding_char = None
+            self._minimum_width = None
+            self.format_spec = None
+            return
+
+        if isinstance(format_spec, basestring) is False:
+            msg = "format_spec for IntegerKey %s is not of type string: %s"
             raise TankError(msg % (name, str(format_spec)))
 
+        if len(format_spec) == 0:
+            raise TankError("format_spec can't be empty.")
+
+        matches = self._FORMAT_SPEC_RE.match(format_spec)
+        if not matches:
+            raise TankError("format_spec for IntegerKey %s has to either be a number (e.g. '3') or "
+                            "a 0 followed by a number (e.g. '03'), not '%s'" % (name, format_spec))
+
+        groups = matches.groups()
+        # groups[0] is either '' or '0', in which case the padding is ' '
+        self._padding_char = groups[0] or ' '
+        # groups[1] is the minimum width of the number.
+        self._minimum_width = int(groups[1])
         self.format_spec = format_spec
 
+    def _init_strict_matching(self, name, strict_matching):
```

Comment:
if we use `self` in the error messages (via `__repr__` as suggested above), we won't need the name param.